iceoryx_posh/include/iceoryx_posh/internal/roudi/memory/mempool_collection_memory_block.hpp

```diff
@@ -45,21 +45,20 @@ class MemPoolCollectionMemoryBlock final : public MemoryBlock
     MemPoolCollectionMemoryBlock& operator=(const MemPoolCollectionMemoryBlock&) = delete;
     MemPoolCollectionMemoryBlock& operator=(MemPoolCollectionMemoryBlock&&) = delete;
 
```

Comment:
As far as I can see the base class `MemoryBlock` does not have a virtual dtor, could you please add one.

Reply:
Since updating the documentation was already not really related to the issue of this PR, are you okay with doing this in a follow up PR? I would also remove the documentation updates from this PR

Reply:
done in #843